components/micro-gateway-cli/src/main/java/org/wso2/apimgt/gateway/cli/utils/CmdUtils.java

```diff
@@ -260,12 +279,131 @@ private static String getProjectTempFolderLocation(String projectName) {
         return getProjectDirectoryPath(projectName) + File.separator + CliConstants.TEMP_DIR_NAME;
     }
 
+    /**
+     * Set api definition file.
+     *
+     * @param projectName name of the project
+     * @param apiDefinition api definition file local path
+     * @param headers api definition download request headers
+     * @param values api definition download values for request headers
+     * @param insecure insecure url connection
+     */
+    private static void setApiDefinition(String projectName, String apiDefinition,
+                                         String headers, String values, boolean insecure) throws IOException {
+        String apiDefinitionsDir = getUserDir() + File.separator + projectName + File.separator +
+                CliConstants.PROJECT_API_DEFINITIONS_DIR + File.separator;
+        String filePath;
+
+        if (isURL(apiDefinition)) {
+            ArrayList<String> headersList = new ArrayList<>();
+            ArrayList<String> valuesList = new ArrayList<>();
+            if (!StringUtils.isBlank(headers) && !StringUtils.isBlank(values)) {
+                headersList = new ArrayList(Arrays.asList(headers.split(",")));
+                valuesList = new ArrayList(Arrays.asList(values.split(",")));
+                if (headersList.size() > 0 && (headersList.size() != valuesList.size())) {
+                    throw new CLIRuntimeException("Provided number of header and number of values is different");
+                }
+                logger.debug("Request headers are provided.");
```

Comment:
Let's add what are the provided headers and the identified values to this log.

Reply:
fixed. thanks